providers/src/airflow/providers/google/cloud/hooks/cloud_sql.py

```diff
@@ -1187,3 +1195,33 @@ def free_reserved_port(self) -> None:
         if self.reserved_tcp_socket:
             self.reserved_tcp_socket.close()
             self.reserved_tcp_socket = None
+
+    def _get_iam_db_login(self) -> str:
+        """Get an IAM login for Cloud SQL database."""
+        if not self.cloudsql_connection.login:
+            raise AirflowException("The login parameter needs to be set in connection")
+
+        if self.database_type == "postgres":
+            return self.cloudsql_connection.login.split(".gserviceaccount.com")[0]
+        else:
```

Comment:
Does "else" cover here only MySQL or all possible other databases in CloudSQL?

If it only covers MySQL, it seems that better design would be to throw an error when use_iam is set to true and database is not in the list of supported.

Reply:
Yes the "else" covers only MySQL. But it is not a problem, because CloudSQL operator works only with two types of databases(MySQL and Postgresql. In the `CloudSQLDatabaseHook` inside the `__init__` method the code runs the `_validate_inputs` function. This function checks that users use one of the valid [DB](https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/google/cloud/hooks/cloud_sql.py#L744) and if not raise an [exception](https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/google/cloud/hooks/cloud_sql.py#L965).

For us it means that the user will have an exception before my code will be executed in case when the wrong database type will be specified.

Reply:
ok, makes sense.

It is a bit unreadable to have this "else" like this, as for reader it looks like for all other database types the logic is here, and you need to find what is the list of other supported databases. Also, once we support other database types, this if/else should be extended.

IMHO, it looks more readable if there would be 
"""
elif self.database_type == "mysql":
    ...
else:
    raise AssertError("Unsupported ...")
"""